### ISSUE TITLE
[Long-term] Keyboard layout autodetection

### ISSUE THREAD

**Issue by ZirconiumX** [ckb-next/ckb-next#89]
# TL;DR
If you have a non-American (ANSI QWERTY English) keyboard layout, would you mind heading to the #ckb-next IRC channel so we can get a packet capture from your device?

# The technical bit
At the moment, the GUI essentially guesses the keyboard layout based on locale, rather than based on the device itself. However, myself and @tatokis think that the keyboard sends the layout as part of its initialisation packet, and so wish to detect and expose this from the daemon to the GUI, so the device works on multiple locales.

## The protocol bit
Initially we thought the device would expose a unique locale-specific layout in a single byte - specifically byte 4 from the device-specific section (currently marked as [keyboard layout identifier]( https://github.com/ckb-next/corsair-protocol/blob/master/host-to-device/0e/01/0e01.md)), but we now think the protocol splits it into a physical layout (ANSI (wide enter), ISO (double height enter), ABNT (double height enter with small right shift), JIS (double height enter with kata/hira keys etc)) and then a virtual layout on top of that (QWERTY, QWERTZ, etc).

## The `/dev` bit
We plan to expose a `/layout` node, which exposes either a layout (layout names currently undecided - I'm in favour of something like `ansi-en` or `jis-jp` while @tatokis suggests just `us` or `jp`) or is empty (in the case of an unknown layout or a non-keyboard device). If a keyboard has an empty `/layout` node, the GUI will prompt the user to file a bug report so we can add it to CKB, and then guess based on locale.
